lib/assessment.js

```diff
@@ -8,38 +8,57 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
-
+var min = function min(previousValue, currentValue, index, array) {
+  return currentValue > previousValue ? currentValue : previousValue;
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-var sum = function sum() {
-
+var sum = function sum(previousValue, currentValue, index, array) {
+  return previousValue + currentValue;
 };
 
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
-var any = function any(array, callback, thisArg) {
 
+// this seems vague, and I don't exactly know
+// what's being asked here, so I'm making a function
+// that uses a for loop to set every item in the array
+// to be 1
+var any = function any(array, callback, thisArg) {
+  for (var i = 0; i < array; i++) {
+    callback(array[i], i, array) = 1;
+  }
+  return true;
 };
 
 // Question 4
 // Write a function that returns a new array
 // made up of the array elements for which invoking
 // the callback returns a truthy value
-var select = function select(array, callback, thisArg) {
 
+// Stil not clear. So, this is what I've got.
+var select = function select(array, callback, thisArg) {
+  var newArray = [];
+  for (var i = 0; i < array; i++) {
+    newArray[i] = array[i]
+  }
+  return true;
```

Comment:
This is close. To do it this way, using a classic `for` loop, you'd need to keep multiple loop indices. On the other hand, if you use `Array.forEach` and `Array.push`, you wouldn't need any index variables.

The problem I pointed out earlier regarding comparing a number to an array also applies here. If this loop were well-formed, it would copy every reference in the source array to the newly-created target array, then lose the reference to the target array and return a boolean instead.

Nonetheless, this is a good start. If you apply one of the approaches mentioned in the first paragraph and use an `if` statement to test for a truthy return value from invoking the callback on the current list element, index, and array, this would be a good answer. 1/2 credit.